aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -523,6 +523,70 @@ void lgamma_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void clamp_kernel_cuda(TensorIterator& iter, Scalar min_scalar, Scalar max_scalar) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX(iter.dtype(), "clamp_cuda", [&]() {
```

Comment:
why did you add complex?  That's not in the TH implementation, right?  And I didn't see it mentioned in the commit message.

Reply:
To my understanding the TH kernel wasn't defined for complex types. I've added complex types in the Aten cuda kernel implementation to match the Aten cpu kernel, which is defined for complex types too (support was added in https://github.com/pytorch/pytorch/commit/7c472ec5974cc638cb697b4523d88324aee38fa9). Would it be preferred to add the support for complex types for the cuda kernel in a separate PR? 

I will split the addition of the cuda kernels into a separate commit and mention the addition of complex types.

Reply:
The complex behaviour in this kernel is equivalent to that in the CPU kernel. Following the current discussion on complex behaviour, should I keep it so that the CPU and CUDA kernels are aligned, or should I remove the complex dispatcher?

Keeping in mind that complex tensors are disabled in `UnaryOps.cpp`, keeping this implementation should be ok?
https://github.com/pytorch/pytorch/blob/5d9b4d5720eff32a6997a66402b6423d703313f6/aten/src/ATen/native/UnaryOps.cpp#L264-L265